frontend/app/src/next/hooks/use-members.tsx

```diff
@@ -92,11 +99,39 @@ export function MembersProvider({ children }: { children: React.ReactNode }) {
     },
   });
 
+  const updateMemberMutation = useMutation({
+    mutationKey: ['tenant-member:update', tenantId],
+    mutationFn: async ({
+      memberId,
+      data,
+    }: {
+      memberId: string;
+      data: UpdateTenantMemberRequest;
+    }) => {
+      try {
+        const res = await api.tenantMemberUpdate(tenantId, memberId, data);
+        return res.data;
+      } catch (error) {
+        toast({
+          title: 'Error updating member role',
+
+          variant: 'destructive',
+          error,
+        });
+        throw error;
+      }
+    },
+    onSuccess: async () => {
+      await membersQuery.refetch();
```

Comment:
I know we've done this in a bunch of places so it's fine here too, but we're doing to need to change our use of `refetch` to `queryClient.invalidateQueries` in many places as some point, so it might be preferable to do that here now so it's less work for the future 😅